apis/externalsecrets/v1beta1/secretsstore_infisical_types.go

```diff
@@ -0,0 +1,55 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	esmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+)
+
+type UniversalAuthCredentials struct {
+	// +kubebuilder:validation:Required
+	ClientID esmeta.SecretKeySelector `json:"clientId"`
+	// +kubebuilder:validation:Required
+	ClientSecret esmeta.SecretKeySelector `json:"clientSecret"`
+}
+
+type InfisicalAuth struct {
+	// Type of machine identity authentication with Infisical
+	// +kubebuilder:validation:Enum=universal-auth
+	Type string `json:"type"`
```

Comment:
Typically, we do not have a `auth.type` in our providers, and instead make the `Auth` structure to only accept one property. This would be easier to maintain code wise

Reply:
I have attached a `type` to make the validation easier actually. So we actually support [multiple kinds of authentication](https://infisical.com/docs/documentation/platform/identities/machine-identities#authentication-methods). 

So there is AWS based authentication which we will add later. Same goes for GCP and k8s based. My thought process was with a required `type` field i can be precise which field inside auth like `universalAuthCreds` or more ones like should I validate. 

May i know what's your take on this and anyway you think that would make it better or improve it?

Reply:
Answered: https://github.com/external-secrets/external-secrets/pull/3477#discussion_r1597456150